### PR TITLE
Added XM51 shotgun

### LINKED ISSUE #3768: XM51 Breaching Scattergun Crate (x2) Cost: $3000

### PR DESCRIPTION
**Pull request by MetalSage**
## About the PR

Close #3768

## Media

https://github.com/user-attachments/assets/25f8c12a-6ed4-4536-9059-044cdde1703b

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added XM51 shotgun

### CODE REVIEW

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml [hunk 1]
```diff
@@ -102,5 +102,29 @@
...
+  parent: CMShellShotgunBase
+  id: RMCShellShotgunBreaching
+  name: handful of breaching shells
+  components:
+  - type: Sprite
+    sprite: _RMC14/Objects/Weapons/Guns/Ammunition/Handfuls/shotgun_handfuls.rsi
+    layers:
+    - state: breaching_shell_1
+  - type: CartridgeAmmo
+    count: 3
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/shotgun.yml:115]: Missing one extra projectile.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/stock_attachments.yml [hunk 2]
```diff
@@ -74,6 +74,48 @@
...
+      accuracyAddMult: 0.2
+      recoilFlat: -2
+      scatterFlat: -6
+    - conditions:
+        unwieldedOnly: true
+      accuracyMovementPenaltyAddMult: -1
+      accuracyAddMult: -0.15
+      recoilFlat: 2
+      scatterFlat: 6
+
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/stock_attachments.yml:118]: Doublecheck this, please. Both of the accuracy modifiers are wrong, as is the unwielded scatter. Burst fire addition is missing. See the BFA in `under_attachments.yml` for how to add it.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml [hunk 3]
```diff
@@ -116,5 +116,93 @@
...
+      path: /Audio/Weapons/Guns/Empty/empty.ogg
+  - type: RMCSelectiveFire
+    baseFireModes:
+    - SemiAuto
+    recoilWielded: 2
+    recoilUnwielded: 4
+    scatterWielded: 10
+    scatterUnwielded: 10
+    baseFireRate: 0.5
+    burstScatterMult: 5
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml:155]: Doublecheck the unwielded scatter and the burst scatter multiplier. The base fire rate also seems to be wrong. I'm getting 0,625 from `1 / (FIRE_DELAY_TIER_4 * 2)`

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml [hunk 4]
```diff
@@ -116,5 +116,93 @@
...
+      gun_magazine:
+        name: Magazine
+        insertSound: /Audio/_RMC14/Weapons/Guns/Reload/m41_reload.ogg
+        ejectSound: /Audio/_RMC14/Weapons/Guns/Reload/m41_unload.ogg
+        priority: 2
+        whitelist:
+          tags:
+          - RMCMagazineShotgunXM51
+  - type: GunDamageModifier
+    multiplier: 1.1
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml:167]: Unless I'm severely overlooking something in the code, the XM51 isn't supposed to have a general damage multiplier. It has a set of multipliers against specific targets.

Set them using `GunDamageMultipliersComponent`. See the U7 underbarrel shotgun in `under_attachments.yml` for how to format them. The flags are in the component file.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/shotgun.yml [hunk 5]
```diff
@@ -95,3 +95,20 @@
...
+  - type: Sprite
+    sprite: _RMC14/Objects/Weapons/Guns/Ammunition/Projectiles/shotgun_projectiles.rsi
+    layers:
+    - state: flechette
+      shader: unshaded
+  - type: Projectile
+    damage:
+      types:
+        Blunt: 25
+        Structural: 500
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/shotgun.yml:112]: Why does this do structural? And blunt? Our bullets normally do piercing. Projectile stats from CM:

projectile

We'll be able to use `RMCExtraProjectilesDamageModsComponent` to adjust the additional projectiles' damage once #3422 is merged.

**Review by Siigiil — CHANGES_REQUESTED** [review]
A bunch of numbers are wrong.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml [hunk 6]
```diff
@@ -116,5 +116,93 @@
       rmc-aslot-stock: -0.375, -0.029
       rmc-aslot-underbarrel: 0.25, -0.31
 
+- type: entity
+  parent: [ CMBaseWeaponRifle, RMCBaseAttachableHolder ]
```
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml:120]: Why does this inherit from the base rifle when it's in the shotgun file? I understand that CM has it under rifles for some bloody reason, but we have separate files for this for a reason. This gun is functionally a shotgun. It should go under shotguns. We can just adjust any stats that changes.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/shotgun.yml (hunk 5 above)
Comment (MetalSage) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/shotgun.yml]: I'm relying on the damage numbers from cm13 and making them approximate, and also as far as I know I can't run 2 different projectiles at once at the same time
image

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/stock_attachments.yml (hunk 2 above)
Comment (MetalSage) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/stock_attachments.yml]: Kinda tweaked the numbers, but I can't add the BFA for the shotgun as it requires pump before firing it

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml (hunk 6 above)
Comment (Dutch-VanDerLinde) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Shotguns/marine_shotguns.yml:120]: you can split the prototype in to two different prototypes (one with the components you dont need, and the other one

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Pull request merged**